contrib/datastore/file/gcp/gcp.go

```diff
@@ -30,7 +44,44 @@ type GcpAdapter struct {
 }
 
 func (g GcpAdapter) CreateFileShare(ctx context.Context, fs *file.CreateFileShareRequest) (*file.CreateFileShareResponse, error) {
-	panic("implement me")
+	instanceName := fs.Fileshare.Name + "-instance"
+	labels := make(map[string]string)
+	for _, tag := range fs.Fileshare.Tags {
+		labels[tag.Key] = tag.Value
+	}
+
+	instance := &gcpfilev1.Instance{
+		Description: fs.Fileshare.Description,
+		Tier: fs.Fileshare.Metadata.Fields[Tier].GetStringValue(),
+		FileShares: []*gcpfilev1.FileShareConfig{
+			{
+				Name:       fs.Fileshare.Name,
+				CapacityGb: fs.Fileshare.Size / utils.GB_FACTOR,
```

Comment:
Why do we expect the input to come in bytes? User can specify simply 1GB or 1 TB etc.

Reply:
I didn't get the point for the input expectation in bytes and user specification? Where do you expect User simply specify GB or TB?
IMHO we should provide the basic storage unit i.e. byte across internal API calls & db persistence and let the respective backend decide to consume in their API specification format, some may consume in bytes, GB, TB etc.

Reply:
Right. Let’s keep it this way. So expectation is that API consumer will do conversions